### PR TITLE
Bump sdl2 mixer to 2.0.4.

### PR DESCRIPTION
**Pull request by matham**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy`
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matham** [comment]
@conda-forge-admin, please rerender

**Comment by matham** [comment]
On appveyor vs2015 it finally worked, but not on vs2008. On azure, neither vs2015 nor vs2008 work.

On appveyor vs2008 fails with:
```
ResourceCompile:
  C:\Program Files\Microsoft SDKs\Windows\v7.0\bin\x64\rc.exe /D NDEBUG /D _VC80_UPGRADE=0x0600 /l"0x0409" /nologo /fo"x64\Release\Version.res" Version.rc 
TRACKER : error TRK0002: Failed to execute command: ""C:\Program Files\Microsoft SDKs\Windows\v7.0\bin\x64\rc.exe" /D NDEBUG /D _VC80_UPGRADE=0x0600 /l0x0409 /nologo /fox64\Release\Version.res Version.rc". The operation identifier is not valid. [%SRC_DIR%\VisualC\SDL_mixer.vcxproj]
```

On azure, for 2015 devenv doesn't seem to be installed. For 2008, it just fails with no clear error, perhaps msbuild is also broken, like on appveyor?.

Not sure what the next step would be.

**Comment by dschreij** [comment]
@matham, the mixer is a tough one. I started an attempt once to upgrade to 2.0.2 (https://github.com/conda-forge/sdl2_mixer-feedstock/pull/2) but didn't have the time to bring it to a successful end. The problem is that mixer comes together with a collection of other sound/codec libraries, of which the composition seems to change with every release of sdl2_mixer. These libraries are also compiled and (statically?) linked by the cmake file (even though I don't know if this is the right way to go) and thus the cmake file needs some work after every new release of sdl2_mixer. 
I don't really know if the problems your are currently bumping into are related to this, but just telling you as a head's up.

**Comment by dschreij** [comment]
Alright, I see that you are not using the cmake file anymore, but using the supplied .sln files instead. That's way better I think.

**Comment by matham** [comment]
It doesn't seem like it common for conda feedstocks to use msbuild tools because there are too many issues. After all that messing around, on azure devenv seems to be installed with 2017 visual studio, which gets stuck at the solution upgrade stage. On appveyor it's installed with vs14, and works properly.

The vs2008 build fails on both appveyor and azure, possibly because the solution is not compatible with it?

I think someone with more experience with ms build needs to take a look at this. I was only able to test locally with vs14 as I don't have 2017 or 2008 installed.

**Comment by dschreij** [comment]
@conda-forge/core could you help with this? Let's keep this on the radar. If the problem really lies with the VS installations, then this should build whenever those are in order.

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
+:: See https://github.com/conda-forge/staged-recipes/pull/194#issuecomment-203577297
+:: Nasty workaround. Need to move a more current msbuild into PATH.  The one on
+:: AppVeyor barfs on the solution. This one comes from the Win7 SDK (.net 4.0),
+:: and is known to work.
+if %VS_MAJOR%==9 (
+    set "PATH=C:\Windows\Microsoft.NET\Framework\v4.0.30319;%PATH%"
+)
 
-:: Configure using the CMakeFiles
-cmake -G "NMake Makefiles" -DCMAKE_INSTALL_PREFIX:PATH="%LIBRARY_PREFIX%" -DCMAKE_BUILD_TYPE:STRING=Release ..
```
Comment (ocefpaf) [inline recipe/bld.bat:10]: I don't understand much about building on Windows but does the `cmake` alternative still exists? It is usually easier to debug.

Comment (dschreij) [reply recipe/bld.bat:10]: I composed that CMakeLists.txt a long time ago, and never updated it to support the newer version of sdl2_mixer. The general concensus of the SDL2 module maintainers is that windows builds are already accounted for by the included .sln files, and therefore they don't feel much for also supporting the cmake build system.

Comment (ocefpaf) [reply recipe/bld.bat:10]: Got it. Sorry but I can't really help you much there. Not sure if the `vc` activation are to blame though. Maybe @isuruf has a better insight.

**Comment by matham** [comment]
I think the issue is that appveyor has visual studio 2015 installed, which comes with the devenv we use to upgrade the solution and then compile it with 2015 msbuild. 

Azure seems to have 2017 instead and its devenv gets stuck for like 10 min trying to upgrade the solution. Furthermore, we then try to compile it with 2015, which is probably why it fails. Azure only has the compiler tools for 2015 and seems to be missing devenv, hence the problem.

**Comment by matham** [comment]
I did some further testing and asked for help here: https://github.com/orgs/conda-forge/teams/core/discussions/3.

**Comment by matham** [comment]
I'm wondering if we should just disable building it with vs2008? That would just break 2.7 installs, which is reaching EOL soon anyway. Specifically, I downloaded vs2008 and tried to open the solution but it said it was created with a later version of visual studio so it couldn't open it.

For 2015, it works on appveyor and I'm hoping https://github.com/conda-forge/conda-smithy/pull/1066 would fix the issue on azure. Would you be ok with dropping vs2008 support, unless someone else figures it out?

**Comment by dschreij** [comment]
Hi @matham. Thanks again for all your contributions. I don't know if it is up to me to drop support for certain platforms, so I'll ask @conda-forge/core to chime in on this. If they are OK with it, I will be too.

Come to speak of it @matham, would you like to be added as a maintainer for all the sdl2 packages? You seem to outskill me regarding build processes (especially on windows) so I'd appreciate your help in keeping this show on the road.

**Comment by ocefpaf** [comment]
> Hi @matham. Thanks again for all your contributions. I don't know if it is up to me to drop support for certain platforms, so I'll ask @conda-forge/core to chime in on this. If they are OK with it, I will be too.

It is up to the maintainers. If supporting a platform is too hard just drop it and, if you want, open an issue asking for help. For that specific case (Windows+Python 2.7) I don't think anyone will care.

**Comment by matham** [comment]
@dschreij I added myself to maintainers. Not sure where the future will take me, but I should be able to help while I'm at my current workplace.

I also disabled building on vs2008 and windows (people on those systems could just remain using the last release). So now the only issue seems to be with zure vs2015, which will hopefully be resolved with https://github.com/conda-forge/conda-smithy/pull/1066.

**Comment by isuruf** [comment]
Packages will be uploaded from appveyor. So even if azure fails, it's fine to merge.

**Pull request merged**